backend/log/context_grpc.go

```diff
@@ -0,0 +1,44 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	logParamsCtxMetadataKey = "loggerParamsCtxMetadata"
+	logParamSeparator       = ":"
+)
+
+// WithContextualAttributesForOutgoingContext will append the given key/value log parameters to the outgoing context.
+func WithContextualAttributesForOutgoingContext(ctx context.Context, logParams []any) context.Context {
+	if len(logParams) == 0 || len(logParams)%2 != 0 {
+		return ctx
+	}
+
+	for i := 0; i < len(logParams); i += 2 {
+		k := logParams[i].(string)
+		v := logParams[i+1].(string)
+		ctx = metadata.AppendToOutgoingContext(ctx, logParamsCtxMetadataKey, fmt.Sprintf("%s%s%s", k, logParamSeparator, v))
+	}
+
+	return ctx
+}
+
+// ContextualAttributesFromIncomingContext returns the contextual key/value log parameters from the given incoming context.
+func ContextualAttributesFromIncomingContext(ctx context.Context) []any {
```

Comment:
Does this need to be public?

Reply:
Aye because it's called from the `backend` package IE `backend/log.go`